engine/render/src/render/render_script.cpp

```diff
@@ -1372,29 +1409,63 @@ namespace dmRender
     int RenderScript_EnableTexture(lua_State* L)
     {
         RenderScriptInstance* i = RenderScriptInstance_Check(L);
-        dmGraphics::HRenderTarget render_target = 0x0;
 
         uint32_t unit = luaL_checknumber(L, 1);
         if (lua_isnumber(L, 2))
         {
-            render_target = (dmGraphics::HRenderTarget) lua_tonumber(L, 2);
-            int buffer_type_value = (int)luaL_checknumber(L, 3);
-            dmGraphics::BufferType buffer_type = (dmGraphics::BufferType) buffer_type_value;
+            dmGraphics::HAssetHandle asset_handle = (dmGraphics::HAssetHandle) lua_tonumber(L, 2);
+            dmGraphics::AssetType asset_type      = dmGraphics::GetAssetType(asset_handle);
+            dmGraphics::HTexture texture          = 0;
 
-            dmGraphics::HTexture texture = dmGraphics::GetRenderTargetTexture(render_target, buffer_type);
+            if (!dmGraphics::IsAssetHandleValid(i->m_RenderContext->m_GraphicsContext, asset_handle))
+            {
+                char buf[128];
+                return luaL_error(L, "Texture handle '%s' is not valid.", AssetHandleToString(asset_handle, buf, sizeof(buf)));
+            }
+
+            if (asset_type == dmGraphics::ASSET_TYPE_RENDER_TARGET)
+            {
+                dmGraphics::BufferType buffer_type = dmGraphics::BUFFER_TYPE_COLOR0_BIT;
+                if (lua_isnumber(L, 3))
```

Comment:
No need to use double for the type I guess?
Integer should be fine?

Reply:
Sure, this is just moving old code I think but I can fix it 👍